gtsam/sfm/DsfTrackGenerator.h

```diff
@@ -0,0 +1,188 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file DsfTrackGenerator.h
+ * @date July 2022
+ * @author John Lambert
+ * @brief Identifies connected components in the keypoint matches graph.
+ */
+
+#pragma once
+#include <Eigen/Core>
+
+#include <gtsam/base/DSFMap.h>
+#include <gtsam/geometry/Point2.h>
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <optional>
+#include <vector>
+
+namespace gtsam {
+
+typedef DSFMap<IndexPair> DSFMapIndexPair;
+typedef std::pair<size_t, size_t> ImagePair;
+typedef Eigen::MatrixX2i CorrespondenceIndices; // N x 2 array
+
+//struct Keypoints;
+using KeypointCoordinates = Eigen::MatrixX2d;
+
+
+struct Keypoints {
+  KeypointCoordinates coordinates;
+  // typedef'd for Eigen::VectorXd
+  boost::optional<gtsam::Vector> scales;
+  boost::optional<gtsam::Vector> responses;
+
+  Keypoints(const gtsam::KeypointCoordinates& coordinates): coordinates(coordinates) {}; // boost::none
+};
+
+using KeypointsList = std::vector<Keypoints>;
+using KeypointsVector = std::vector<Keypoints>; // TODO(johnwlambert): prefer KeypointsSet?
```

Comment:
do we need both?

Reply:
Wasn't sure what our current convention for this is. I think `KeypointsVector` could create some confusion since it's a vector of keypoint vectors (2d coordinates). So I'd prefer `KeypointsList` or `KeypointsSet`.

Reply:
@dellaert @akshay-krishnan any votes on the naming between KeypointsVector vs KeypointsList?

Reply:
either works for me, I think these names are getting pretty confusing anyway. Ideally APIs using std::vector<Keypoints> should be implicitly converted to a list in python without needing a name for it :)

Reply:
Agreed. @ProfFan @dellaert is the use of stl.h's implicit conversion something on the roadmap or is it too bug-prone?